lms/views/basic_lti_launch.py

```diff
@@ -89,7 +90,7 @@ def store_lti_data(self):
         if not lti_user.is_instructor and not self.context.is_canvas:
             # Create or update a record of LIS result data for a student launch
             request.find_service(name="grading_info").upsert_from_request(
-                request, h_user=lti_user.h_user, lti_user=lti_user
+                request, lti_user.h_user, lti_user.user_id, self.application_instance
             )
```

Comment:
Eh... the number of parameters is kind of bugging me here and in the tests. I know it's more explicit, but our app is riddled with global state on the request object. I feel like we should either accept the individual arguments (and not the request) or just use the request.

If we simplify this code down it's:

```python
request.find_service(name="grading_info").upsert_from_request(
    request, 
    request.lti_user.h_user, 
    request.lti_user.user_id, 
    request.find_service(name="application_instance").get_current()
)
```

Which feels like it could be:

```python
request.find_service(name="grading_info").upsert_from_request(request)
```